### PR TITLE
ENYO-2153: Add user defined component to accessibilitySample

### PR DESCRIPTION
**Pull request by kbs12e**
Accessibility sample is  for reference unusual cases.
Thus, we classified each case and added user defined sample.
In addition, reload button is removed because reload is already in right-top side of all moonstone samples.

Enyo-DCO-1.1-Signed-off-by: Bongsub Kim bongsub.kim@lgepartner.com

### CODE REVIEW

File: src/moonstone-samples/lib/AccessibilitySample.js [hunk 1]
```diff
@@ -23,32 +24,56 @@ module.exports = kind({
...
+			{kind: Icon, src: '@../assets/icon-list.png', ontap: 'buttonTapped'},
+			{classes: 'moon-1v'},
+			{kind: Control, tag: 'br'},
+			{kind: Divider, content: 'Case 2: User defined'},
+			{kind: Item, classes: 'moon-hspacing', ontap: 'buttonTapped', components: [
+				{content: 'Content and Icon'},
 				{kind: Icon, src: '@../assets/icon-list.png', ontap: 'buttonTapped'}
 			]},
-			{kind: Control, tag: 'br'},
+			{name: 'checkItem', kind: Item, classes: 'moon-hspacing', ontap: 'buttonTapped', components: [
```
Comment (ryanjduffy) [inline src/moonstone-samples/lib/AccessibilitySample.js]: If the goal here is to represent how to make a custom kind that is accessible, it should be defined as such separate from the sample. That will give us the opportunity to show the accessibility-specific members (e.g. `ariaObservers`, `accessibilityRole`, etc).

File: src/moonstone-samples/lib/AccessibilitySample.js [hunk 2]
```diff
@@ -23,32 +24,56 @@ module.exports = kind({
...
 			{classes: 'moon-1v'},
-			{kind: Divider, content: 'Icons: '},
-			{kind: Icon, icon: 'play', small: false, ontap: 'buttonTapped', accessibilityLabel: 'play'},
-			{kind: Icon, src: '@../assets/icon-list.png', ontap: 'buttonTapped', accessibilityLabel: 'icon List'},
-			{kind: Item, classes: 'moon-hspacing', ontap: 'buttonTapped', accessibilityLabel: 'selectable icon List', components: [
-				{content: 'Selectable Item', accessibilityDisabled: true},
+			{kind: Control, tag: 'br'},
+			{kind: Divider, content: 'Case 1: Non spotlight'},
+			{kind: Icon, icon: 'play', small: false, ontap: 'buttonTapped'},
+			{kind: Icon, src: '@../assets/icon-list.png', ontap: 'buttonTapped'},
```
Comment (ryanjduffy) [inline src/moonstone-samples/lib/AccessibilitySample.js:68]: Depending on how ENYO-2430 plays out, we might need to add `tabIndex` here to illustrate that property for non-spottable controls.

File: src/moonstone-samples/lib/AccessibilitySample.js [hunk 3]
```diff
@@ -23,32 +24,56 @@ module.exports = kind({
...
+				{name: 'checkbox', kind: Checkbox, spotlight:false, checked: true}
+			]},
 			{classes: 'moon-1v'},
-			{kind: Divider, content: 'Icon Buttons: '},
-			{kind: IconButton, icon: 'drawer', small: false, ontap: 'buttonTapped', accessibilityLabel: 'drawer'},
-			{kind: IconButton, src: '@../assets/icon-list.png', small: false, ontap: 'buttonTapped', accessibilityLabel: 'icon List'}
+			{kind: Control, tag: 'br'},
+			{kind: Divider, content: 'Case 3: Non content'},
+			{kind: IconButton, icon: 'drawer', small: false, ontap: 'buttonTapped'},
+			{kind: IconButton, src: '@../assets/icon-list.png', small: false, ontap: 'buttonTapped'}
```
Comment (ryanjduffy) [inline src/moonstone-samples/lib/AccessibilitySample.js]: Why did you remove `accessibilityLabel` from these? I think we'd want to show that here.

File: src/moonstone-samples/lib/AccessibilitySample.js (hunk 2 above)
Comment (kbs12e) [inline src/moonstone-samples/lib/AccessibilitySample.js:68]: Yes. My intent is that default Icon can not read any string when clicked, but can read label or hint string if accessibilityLabel or accessibilityHint is added.

File: src/moonstone-samples/lib/AccessibilitySample.js (hunk 1 above)
Comment (kbs12e) [inline src/moonstone-samples/lib/AccessibilitySample.js]: Ok. I updated it using accessibilityRole and ariaObservers

File: src/moonstone-samples/lib/AccessibilitySample.js (hunk 3 above)
Comment (kbs12e) [inline src/moonstone-samples/lib/AccessibilitySample.js]: Ok. I reverted this lines.
